### ISSUE TITLE
Implement action allowing to ignore following results in CLI for rest of file

### ISSUE THREAD

**Issue by Makman2** [coala/coala#4592]

**Comment by Makman2** [comment]
I'm not sure the current action subsystem allows this^^

**Comment by Makman2** [comment]
CC @sils

**Comment by shreyans800755** [comment]
You're right. There's no option to ignore the actions. We should have something like that.

**Comment by shreyans800755** [comment]
But there is something called `IgnoreResultAction` which allows user to add comment on erroneous part, by asking language to choose how to add the comment.

**Comment by Makman2** [comment]
But it works on a single result. My approach involves control over all (following) results, and this is the questionable part^^

**Comment by Makman2** [comment]
Btw there's already an ignore-action which does this but for a single result only ;)

**Comment by shreyans800755** [comment]
@Makman2 
> Btw there's already an ignore-action which does this but for a single result only ;)

I can't see that option while running `coala`.
I only get these 3 options:

> 0. Do (N)othing
> 1. (O)pen file
> 2. Add (I)gnore comment

**Comment by Makman2** [comment]
> 0. Do (N)othing

that's it :)

**Comment by shreyans800755** [comment]
It still applies the patch, though.

**Comment by Makman2** [comment]
> It still applies the patch, though.

What patch? :O

**Comment by shreyans800755** [comment]
In case of `ApplyPatchAction`, even though we select `Do Nothing`, it applies the patch/fix.

**Comment by Makman2** [comment]
Not for me. However, file an issue, sounds like a bug.

**Comment by mrtes** [comment]
Can the file be extracted from the result?
If so, we could pass in the whole list of results to the action handling sequence and filter all result in that file if the action is chosen.
Or return something from `acquire_actions_and_apply` and `print_result`, that allows filtering.

Or extract all the interactive stuff in one or more classes.
This would allow us to store state that is currently passed through a lot of functions (often unused) as instance variables and make the function signatures more comprehensible. 
If done correctly, this should also allow us to implement new features, like this action, more easily.

Obviously, that needs more thought, but it could be a worthwhile change.

**Comment by Makman2** [comment]
> If so, we could pass in the whole list of results to the action handling sequence and filter all result in that file if the action is chosen.

Problem is, that happens one by one, as soon as they arrive ;)
We would need some kind of "control signals", like an action is chosen as being "active". It receives results it can apply actions for. If it's done, it somehow "returns".

**Comment by mrtes** [comment]
Ah, that wasn't clear to me. I thought they are collected all at once, or section-wise.
Are there docs, that explain the process of how results are collected?

Anyway, If it works like a queue or stream (which I'm assuming right now) , the action could set some sort of filter, that removes all results matching that file, as they come?

**Comment by Makman2** [comment]
> Are there docs, that explain the process of how results are collected?

Not sure, don't think it's officially documented, but it's officially wanted :D (see `Interactions.py` which is a mess right now. I'm currently on it to replace it with the nextgen-core design, however it's merging very slowly because nearly nobody reviews it :/)

> Anyway, If it works like a queue or stream (which I'm assuming right now) , the action could set some sort of filter, that removes all results matching that file, as they come?

That's a nice idea. If possible I also want to allow to remove those filters somehow again, otherwise it's a bit limited, but this approach should be quite easy.
